File: stress-test/agg_perseus.R
suppressMessages({library(readr); library(dplyr)})
D <- "/fs/scratch/PUOM0008/crsfaaron/fvs_stress/out_perseus_wo1"
fs <- list.files(D, pattern="^perseus_100yr_agb_batch.*\\.csv$", full.names=TRUE)
cat("files:", length(fs), "\n")
dat <- bind_rows(lapply(fs, function(f) suppressMessages(read_csv(f, show_col_types=FALSE))))
cat("rows:", nrow(dat), "\n")
write_csv(dat, file.path(D, "perseus_100yr_agb_all.csv"))
summ <- dat %>% group_by(YEAR, VARIANT, CONFIG) %>%
  summarise(n_plots=n_distinct(PLOT), mean_agb_tons_ac=mean(AGB_TONS_AC, na.rm=TRUE), .groups="drop") %>%
  arrange(VARIANT, CONFIG, YEAR)
write_csv(summ, file.path(D, "perseus_100yr_agb_summary.csv"))
cat("=== summary rows:", nrow(summ), "===\n")
print(as.data.frame(summ), row.names=FALSE)

File: stress-test/agg_perseus.py
import glob, pandas as pd
D="/fs/scratch/PUOM0008/crsfaaron/fvs_stress/out_perseus_wo1"
fs=sorted(glob.glob(D+"/perseus_100yr_agb_batch*.csv"))
print("files:",len(fs))
df=pd.concat((pd.read_csv(f) for f in fs), ignore_index=True)
print("rows:",len(df),"cols:",list(df.columns))
df.to_csv(D+"/perseus_100yr_agb_all.csv", index=False)
s=(df.groupby(["YEAR","VARIANT","CONFIG"])
     .agg(n_plots=("PLOT","nunique"), mean_agb_tons_ac=("AGB_TONS_AC","mean"))
     .reset_index().sort_values(["VARIANT","CONFIG","YEAR"]))
s.to_csv(D+"/perseus_100yr_agb_summary.csv", index=False)
print("summary rows:",len(s))
print(s.to_string(index=False))
